public_html/wp-content/plugins/wordcamp-reports/views/public/payment-activity.php

```diff
@@ -47,7 +47,14 @@
 
 		<div class="field_wordcamp-id">
 			<label for="wordcamp-id">WordCamp <span>(optional)</span></label>
-			<?php echo get_wordcamp_dropdown( 'wordcamp-id', array(), $wordcamp_id ); ?>
+			<?php if ( ! isset( $_GET[ 'action' ] ) ) {
+				// Show only latest 100 WordCamps by default
```

Comment:
Pre-COVID, we had around 150 camps per year, so I'd lean towards something more like `300` or `400`, which should still be totally fine for memory. 

We only recently started noticing the memory warnings, and it's not running out of memory yet, so I'm hesitant to make it harder for folks to access the data. I'm guessing we could do something like `800` safely, and it may be better to make the minimal necessary impact.

Reply:
> only recently started noticing the memory warnings, and it's not running out of memory yet

that's fair, although i'm always a little nervous when it almost reaches the limit, and could be knocked over by another ~100 WordCamps :)

Reply:
Yeah, that's fair, it'd be good to do something now so that it doesn't get worse 👍🏻 

Right now it seems like some of the other approaches are better (reducing the data, caching, fetching async), but if we have to then limiting the # of camps could work.